src/Installer/dnup/IDotnetInstaller.cs

```diff
@@ -1,13 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-
-using System.Collections.Generic;
-
-namespace Microsoft.DotNet.Tools.Bootstrapper
-{
-    internal interface IDotnetInstaller
```

Comment:
I'm tentative on removing this interface - I think the IDotnetInstaller interface with install, uninstall is a more helpful layer of abstraction for the library. However, I think this abstraction is helpful for dnup because it separates the steps of installation from one that doesn't need a mutex and the step that does. That could allow us to reuse all of the logic for an admin based installer in the future. (prepare would be downloading the exe/verifying it and commit would be running the installer.)

With that said, I wouldn't consider this blocking, it's very much a future concern and it's not super difficult to address.

Reply:
Thanks, that makes sense to me.